pkg/api/types.go

```diff
@@ -403,10 +403,22 @@ const (
 	ClaimBound PersistentVolumeClaimPhase = "Bound"
 )
 
+type PropagationMode string
+
+const (
+	PropagationPrivate  PropagationMode = "private"
+	PropagationRPrivate PropagationMode = "rprivate"
+	PropagationSlave    PropagationMode = "slave"
+	PropagationRSlave   PropagationMode = "rslave"
+	PropagationShared   PropagationMode = "shared"
+	PropagationRShared  PropagationMode = "rshared"
+)
+
```

Comment:
I think based on the discussion here, the next step is to figure out the answer to the following two questions:

1) Can this be solved by a single sane default for volume mounts?
2) If not 1, what's the minimal feature we need to expose to users to enable people the most, and what's a thinner API for it.

It's possible that this *is* the correct minimal thing to expose, but based on the discussion here, it needs to justify itself (and I personally think it is not).

Based on @thockin's comment and concerns (which I share), I think the next step here is to write a proposal explaining what the options are, what the use cases are, and explaining exactly why a given option is the right one.
I think iterating on a .md proposal will be more productive than iterating on the exact code / api-changes.